docs/source/main_classes/deepspeed.rst

```diff
@@ -1178,9 +1178,10 @@ Since `"auto"` is used the :class:`~transformers.Trainer` arguments will set the
 file. This is so that there is one definitive source of the values and to avoid hard to find errors when, for example,
 the learning rate is set to different values in different places. Command line rules. The values that get set are:
 
-- ``warmup_min_lr`` with the value of ``0``
-- ``warmup_max_lr`` with the value of ``--learning_rate``
-- ``warmup_num_steps`` with the value of ``--warmup_steps``
+- ``warmup_min_lr`` with the value of ``0``.
+- ``warmup_max_lr`` with the value of ``--learning_rate``.
+- ``warmup_num_steps`` with the value of ``--warmup_steps`` if provided. Otherwise will use ``--warmup_ratio``'s value
+  using ``math.ceil(warmup_ratio*total_num_steps)``.
```

Comment:
```suggestion
- ``warmup_num_steps`` with the value of ``--warmup_steps`` if provided. Otherwise will use ``--warmup_ratio`` multiplied by the number of training steps.
```

Reply:
if you prefer not to use code, should it at say "multiplied by the number of training steps and rounded up"? or do you suggest that it's too much detail?

Reply:
We can add rounded up to be more precise, yes.

Reply:
re-edited